diff.go

```diff
@@ -316,16 +361,17 @@ func (d *BundleDiff) Empty() bool {
 
 // ApplicationDiff stores differences between an application in a bundle and a model.
 type ApplicationDiff struct {
-	Missing     DiffSide              `yaml:"missing,omitempty"`
-	Charm       *StringDiff           `yaml:"charm,omitempty"`
-	Series      *StringDiff           `yaml:"series,omitempty"`
-	Placement   *StringDiff           `yaml:"placement,omitempty"`
-	NumUnits    *IntDiff              `yaml:"num_units,omitempty"`
-	Scale       *IntDiff              `yaml:"scale,omitempty"`
-	Expose      *BoolDiff             `yaml:"expose,omitempty"`
-	Options     map[string]OptionDiff `yaml:"options,omitempty"`
-	Annotations map[string]StringDiff `yaml:"annotations,omitempty"`
-	Constraints *StringDiff           `yaml:"constraints,omitempty"`
+	Missing          DiffSide                       `yaml:"missing,omitempty"`
+	Charm            *StringDiff                    `yaml:"charm,omitempty"`
+	Series           *StringDiff                    `yaml:"series,omitempty"`
+	Placement        *StringDiff                    `yaml:"placement,omitempty"`
+	NumUnits         *IntDiff                       `yaml:"num-units,omitempty"`
```

Comment:
You can't do this though, can you, as in you can't change this because you break any diffs for other juju versions... so I think the fix is to leave this as `num_units`, but change `exposed-endpoints`.

Reply:
Not sure if that's the case though since this is a utility package which is linked into the CLI. I don't believe there is an API call that gives you back a diff.

Also, other fields in the diff.go file use dashes with the exception of num-units. Just to be safe, I will use an underscore for exposed endpoints but I will retain the dash for the `ExposedEndpointDiff` struct as we do with RelationDiff and friends.